pkg/controllers/systemd-etcd.go

```diff
@@ -0,0 +1,79 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/openshift/microshift/pkg/config"
+	"github.com/openshift/microshift/pkg/util/cryptomaterial"
+	"k8s.io/klog/v2"
+)
+
+type ExtEtcdService struct {
+	CertFile      string
+	KeyFile       string
+	TrustedCAFile string
+}
+
+func NewExtEtcd(cfg *config.MicroshiftConfig) *ExtEtcdService {
+	s := &ExtEtcdService{}
+	s.configure(cfg)
+	return s
+}
+
+func (s *ExtEtcdService) Name() string           { return "etcd" }
+func (s *ExtEtcdService) Dependencies() []string { return []string{} }
+
+func (s *ExtEtcdService) configure(cfg *config.MicroshiftConfig) {
+	certsDir := cryptomaterial.CertsDirectory(cfg.DataDir)
+
+	caCertFile := cryptomaterial.UltimateTrustBundlePath(certsDir)
+	etcdCertDir := filepath.Join(certsDir, s.Name())
+
+	s.CertFile = filepath.Join(etcdCertDir, "etcd-serving.crt")
+	s.KeyFile = filepath.Join(etcdCertDir, "etcd-serving.key")
+	s.TrustedCAFile = caCertFile
+}
+
+func (s *ExtEtcdService) Run(ctx context.Context, ready chan<- struct{}, stopped chan<- struct{}) error {
+	defer close(stopped)
+
+	wd, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("%s failed to get workdir: %v", s.Name(), err)
+	}
+
```

Comment:
Maybe, I'm currently working on getting the etcd systemd unit tied to the MicroShift unit such that if the latter is killed, etcd comes down as well.

If we'd be more comfortable, for sanity sake, issuing the stop command for microshift-etcd just in case, I'm fine to add it.

Reply:
With the most recent update, the etcd unit will go away when MS goes away (even if MS dies or is killed)